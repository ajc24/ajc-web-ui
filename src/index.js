/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
