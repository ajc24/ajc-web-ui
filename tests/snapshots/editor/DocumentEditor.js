import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { DocumentEditor } from '../../../src';

jest.useFakeTimers();

describe('DocumentEditor', () => {
  describe('Default props and rendering - Component with grey theme', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DocumentEditor />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with all custom properties set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DocumentEditor colour="red" dialogContentAreaColour="grey" reRenderAllowance={500} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Editor displaying screenshot with caption item', () => {
    let jsonSnapshot;
    let querySelectorSpy;

    beforeAll(() => {
      querySelectorSpy = jest
        .spyOn(global.document, 'querySelector')
        .mockImplementation(() => {
          return {
            textContent: 'screenshot-with-caption',
          };
        });
      const wrapper = TestDev.mount(
        <React.Fragment>
          <DocumentEditor />
        </React.Fragment>
      );
      wrapper.instance().addEditorItem();
      wrapper.update();
      const wrapperHtml = TestDev.html(wrapper);
      jsonSnapshot = TestDev.createSnapshot(wrapperHtml);
    });

    afterAll(() => {
      querySelectorSpy.mockRestore();
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
