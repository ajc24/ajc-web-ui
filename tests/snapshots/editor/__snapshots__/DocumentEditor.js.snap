// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentEditor Default props and rendering - Component with grey theme verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-width-hd-inner ajc-spacing-bottom-small ajc-text-black ajc-font-default"
>
  <div
    className="ajc-visibility-visible"
    id="main-editor-container"
  >
    <form
      id="main-editor-form-id"
    />
    <div
      className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left"
    >
      <h2
        className="ajc-font-paragraph-heading"
      >
        Click the button below to add new items to your document:
      </h2>
    </div>
    <div
      className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left ajc-spacing-top-reduced"
    >
      <button
        className="ajc-button ajc-font-default ajc-button-grey"
        id="add-new-item-button"
        onClick={[Function]}
        title="Add a new item to your document."
        type="button"
      >
        Add New Item
      </button>
    </div>
    <span
      className="ajc-display-none"
      id="add-new-item-user-value"
    />
  </div>
  <div
    className="ajc-visibility-hidden"
    id="preview-panel-container"
  >
    Hello World
  </div>
</div>
`;

exports[`DocumentEditor Transferred props and rendering - Component with all custom properties set verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-width-hd-inner ajc-spacing-bottom-small ajc-text-black ajc-font-default"
>
  <div
    className="ajc-visibility-visible"
    id="main-editor-container"
  >
    <form
      id="main-editor-form-id"
    />
    <div
      className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left"
    >
      <h2
        className="ajc-font-paragraph-heading"
      >
        Click the button below to add new items to your document:
      </h2>
    </div>
    <div
      className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left ajc-spacing-top-reduced"
    >
      <button
        className="ajc-button ajc-font-default ajc-button-red"
        id="add-new-item-button"
        onClick={[Function]}
        title="Add a new item to your document."
        type="button"
      >
        Add New Item
      </button>
    </div>
    <span
      className="ajc-display-none"
      id="add-new-item-user-value"
    />
  </div>
  <div
    className="ajc-visibility-hidden"
    id="preview-panel-container"
  >
    Hello World
  </div>
</div>
`;

exports[`DocumentEditor Transferred props and rendering - Editor displaying screenshot with caption item verifies the snapshot for the component 1`] = `"<div class=\\"ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-width-hd-inner ajc-spacing-bottom-small ajc-text-black ajc-font-default\\"><div id=\\"main-editor-container\\" class=\\"ajc-visibility-visible\\"><form id=\\"main-editor-form-id\\"><div id=\\"screenshot-with-caption-0-container\\"><div id=\\"screenshot-with-caption-0-file-selection-root\\" class=\\"ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-black-text ajc-font-default\\"><label for=\\"screenshot-with-caption-0-file\\" class=\\"ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced ajc-label-standalone ajc-spacing-top-large\\">Add / Edit Screenshot 1</label><p class=\\"ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-left\\">Loading. Please wait.</p><img id=\\"screenshot-with-caption-0-image\\" role=\\"presentation\\"><input id=\\"screenshot-with-caption-0-file\\" name=\\"screenshot-with-caption-0-file\\" type=\\"file\\" class=\\"ajc-display-none\\"></div><div id=\\"screenshot-with-caption-0-text-input-root\\" class=\\"ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-spacing-bottom-small ajc-text-black ajc-font-default\\">Text Input Component</div></div></form><div class=\\"ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left ajc-spacing-top-large\\"><h2 class=\\"ajc-font-paragraph-heading\\">Click the button below to add new items to your document:</h2></div><div class=\\"ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left ajc-spacing-top-reduced\\"><button id=\\"add-new-item-button\\" class=\\"ajc-button ajc-font-default ajc-button-grey\\" type=\\"button\\" title=\\"Add a new item to your document.\\">Add New Item</button></div><span id=\\"add-new-item-user-value\\" class=\\"ajc-display-none\\"></span></div><div id=\\"preview-panel-container\\" class=\\"ajc-visibility-hidden\\">Hello World</div></div>"`;
